### PR TITLE
fully type check asyncio module

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

File: mpd/asyncio.py [hunk 1]
```diff
@@ -20,7 +20,23 @@
```
Comment (Mic92) [inline mpd/asyncio.py:20]: Please have a look @chrysn

File: mpd/asyncio.py [hunk 2]
```diff
@@ -97,22 +116,22 @@ class CommandResultIterable(BaseCommandResult):
...
+        self.__spooled_lines: asyncio.Queue[Optional[str]] = asyncio.Queue()
 
-    def _feed_line(self, line):
+    def _feed_line(self, line: Optional[str]) -> None:
         self.__spooled_lines.put_nowait(line)
 
     _feed_error = _feed_line
 
-    def __await__(self):
+    def __await__(self) -> Generator[Any, None, Any]:
```
Comment (Mic92) [inline mpd/asyncio.py:128]: I think this needs a fix.

File: mpd/asyncio.py [hunk 3]
```diff
@@ -97,22 +116,22 @@ class CommandResultIterable(BaseCommandResult):
...
         asyncio.Task(self.__feed_future())
         return super().__await__()
 
     __iter__ = __await__  # for 'yield from' style invocation
 
-    async def __feed_future(self):
+    async def __feed_future(self) -> None:
```
Comment (Mic92) [inline mpd/asyncio.py:134]: And this.

**Comment by chrysn** [comment]
Anything in particular I should (have -- sorry, missed the ping) look(ed) at?

From reading through it it appears fine, and the example runs fine in python3 and mypy doesn't complain about it.

**Pull request closed without merging**